File: articles/warren-mcculloch-walter-pitts-and-neural-logic/java/ThresholdUnit.java
public class ThresholdUnit {
    static int thresholdUnit(int[] inputs, int[] weights, int threshold) {
        int total = 0;
        for (int i = 0; i < inputs.length; i++) total += inputs[i] * weights[i];
        return total >= threshold ? 1 : 0;
    }

    public static void main(String[] args) {
        System.out.println(thresholdUnit(new int[]{1, 1}, new int[]{1, 1}, 2));
    }
}

File: articles/warren-mcculloch-walter-pitts-and-neural-logic/rust/threshold_unit.rs
fn threshold_unit(inputs: &[i32], weights: &[i32], threshold: i32) -> i32 {
    let total: i32 = inputs.iter().zip(weights.iter()).map(|(x, w)| x * w).sum();
    if total >= threshold { 1 } else { 0 }
}

fn main() {
    println!("{}", threshold_unit(&[1, 1], &[1, 1], 2));
}
